WordPress/src/main/java/org/wordpress/android/ui/posts/services/PostUploadService.java

```diff
@@ -132,6 +135,27 @@ public static boolean isPostUploading(PostModel post) {
         return false;
     }
 
+    /**
+     * Returns true if the passed post is currently uploading.
+     * Except for legacy mode, a post counts as 'uploading' if the post content itself is being uploaded - a post
+     * waiting for media to finish uploading counts as 'waiting to be uploaded' until the media uploads complete.
+     */
+    public static boolean isPostUploading(PostModel post) {
+        return mCurrentUploadingPost != null && mCurrentUploadingPost.getId() == post.getId();
+    }
+
+    public static void cancelQueuedPostUpload(PostModel post) {
+        synchronized (mPostsList) {
+            Iterator<PostModel> iterator = mPostsList.iterator();
```

Comment:
Would it be better to rename `mPostsList` to  something like `mQueuedPostsList` now that this PR makes a clearer distinction between something being currently uploaded as opposed to something being in a _queue_ to upload?